prime-router/src/main/kotlin/cli/tests/BasicTests.kt

```diff
@@ -115,126 +137,194 @@ class End2EndUniversalPipeline : CoolTest() {
-            passed = false
+            bad("***$name test FAILED***: Response code $responseCode")
         } else {
-            good("Posting of async report succeeded with response code $responseCode")
+            good("Posting of report succeeded with response code $responseCode")
         }
 
-        echo(json)
-        passed = passed and examinePostResponse(json, false)
-        if (!passed) {
-            bad("***async end2end_up FAILED***: Error in post response")
+        if (!examinePostResponse(json, false)) {
+            bad("***$name test FAILED***: Error in post response")
         }
-        // gets back 'received' reportId
-        val reportId = getReportIdFromResponse(json)
 
-        if (reportId != null) {
-            // check convert step
-            val convertResults = pollForStepResult(reportId, TaskAction.convert)
-            // verify each result is valid
-            for (result in convertResults.values)
-                passed = passed && examineStepResponse(result, "convert", sender.topic)
-            if (!passed) {
-                bad("***async end2end_up FAILED***: Convert result invalid")
-            }
+        return getReportIdFromResponse(json)
+    }
 
-            // check route step
-            val convertReportId = getSingleChildReportId(reportId)
-                ?: return bad("***async end2end_up FAILED***: Convert report id null")
+    /**
+     * Calls the report history endpoint periodically, waiting for the overall status to be set to Delivered.
+     * Times out after 90 seconds.
+     */
+    private suspend fun pauseForBatchProcess(environment: Environment, reportId: ReportId?): String {
+        val maxPollSecs = 90
+        val pollSleepSecs = 5
+        var timeElapsedSecs = 0
+        var overallStatus = ""
 
-            val routeResults = pollForStepResult(convertReportId, TaskAction.route)
-            // verify each result is valid
-            for (result in routeResults.values)
-                passed = passed && examineStepResponse(result, "route", sender.topic)
-            if (!passed) {
-                bad("***async end2end_up FAILED***: Route result invalid")
-            }
-            val receiverNames = routeResults.values.flatMap { it?.reports ?: emptyList() }
-                .map { "${it.receivingOrg}.${it.receivingOrgSvc}" }
-            expectedReceivers.forEach { receiver ->
-                if (!receiverNames.contains(receiver.fullName)) {
-                    bad("***async end2end_up FAILED***: expected ${receiver.fullName} to have received a report")
-                    passed = false
-                }
+        echo("\nPolling for history endpoint for report ID: $reportId. (Max poll time $maxPollSecs seconds)")
+        while (timeElapsedSecs <= maxPollSecs) {
+            delay(pollSleepSecs.toLong() * 1000)
+            timeElapsedSecs += pollSleepSecs
+
+            val getUrl = "${environment.url}/api/waters/report/$reportId/history"
```

Comment:
So this call is returning a 401 in staging.

Reply:
I fixed this by adding an access token that is generated when logging into staging from the Prime CLI https://github.com/CDCgov/prime-reportstream/pull/13679/files?w=0#diff-4025d6168d38b38532a410c9be4bcfa26e83dfb2239af85cf8569c57a77374a8R163. Also swapped out a different HTTP util call.